jvm-packages/xgboost4j-flink/src/main/scala/ml/dmlc/xgboost4j/scala/flink/XGBoostModel.scala

```diff
@@ -17,32 +17,51 @@
 package ml.dmlc.xgboost4j.scala.flink
 
 import ml.dmlc.xgboost4j.LabeledPoint
+import ml.dmlc.xgboost4j.java.flink.{XGBoostModel => JXGBoostModel}
+import ml.dmlc.xgboost4j.java.{XGBoostError, Booster => JBooster}
 import ml.dmlc.xgboost4j.scala.{Booster, DMatrix}
+import org.apache.flink.api.scala.{DataSet, createTypeInformation}
+import org.apache.flink.ml.linalg.{SparseVector, Vector}
 
-import org.apache.flink.api.scala.{DataSet, _}
-import org.apache.flink.ml.math.Vector
-import org.apache.hadoop.conf.Configuration
-import org.apache.hadoop.fs.{FileSystem, Path}
+object XGBoostModel{
+  def apply(booster: JBooster): XGBoostModel = {
+    new XGBoostModel(new JXGBoostModel(booster))
+  }
+  def apply(booster: Booster): XGBoostModel = {
+    new XGBoostModel(new JXGBoostModel(booster.booster))
+  }
+}
 
-class XGBoostModel (booster: Booster) extends Serializable {
+class XGBoostModel(wrapped: JXGBoostModel) extends Serializable {
```

Comment:
I think the name `booster` is fine and seems clearer than the new name `wrapped`?

Reply:
@trivialfis  I am wrapping the Java XGBoostModel implementation here, hence, the naming. I can rename this to javaXGBoostModel or something alike. Do you have a better name in mind? Sadly, booster is not what is used. 

In principle, I delegate all operations made by scala's XGBoostModel to the 'wrapped' java implementation hence the naming choice.

Reply:
I have removed the class and rewrote the scala example so that the java implementation is used.